gpt_index/vector_stores/pinecone.py

```diff
@@ -74,7 +80,28 @@ def __init__(
             import pinecone  # noqa: F401
         except ImportError:
             raise ImportError(import_err_msg)
-        self._pinecone_index = cast(pinecone.Index, pinecone_index)
+
+        self._index_name = index_name
+        self._environment = environment
+        if pinecone_index is not None:
+            self._pinecone_index = cast(pinecone.Index, pinecone_index)
+            _logger.warn(
+                "When directly passing in client, cannot automatically reconstruct connetion after save_to_disk/load_from_disk.\
```

Comment:
if the user wants to specify pinecone_index directly, can they even use save_to_disk/load_from_disk at all? 

since you call `load_additional_query_context_from_dict` in `load_from_disk`, so the user can't override with their own additional_query_context containing the pinecone_index if they wanted to right?

Reply:
Now possible.